### ISSUE TITLE
Analyzer Configuration for CA1051,CA1826 not available in .NET5.0

### ISSUE THREAD

**Issue by wzchua** [dotnet/roslyn-analyzers#5188]
### Analyzer

**Diagnostic ID**: CA1051,CA1826, maybe some others

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.301](https://dotnet.microsoft.com/download/dotnet/5.0)

### Describe the bug
Analyzer Configuration is not taken into account

### Steps To Reproduce

1. create a new console project.
2. Add .editorconfig
```
root = true

[*.cs]
dotnet_diagnostic.CA1051.severity = warning
dotnet_code_quality.exclude_structs = true
```
3. Add a struct with a public field to the project
```cs
public readonly struct S
{
    public readonly int Field;
}
```
4. dotnet build
### Expected behavior

No warnings

### Actual behavior

Warning CA1051

**Comment by Youssef1313** [comment]
@wzchua I cannot reproduce the issue. The warning goes away with `dotnet_code_quality.exclude_structs = true` (also this is unit tested). It could be you're on a version that doesn't include the fix from https://github.com/dotnet/roslyn-analyzers/pull/4162.

**Comment by wzchua** [comment]
You're correct.
I did a test with a github action .NET 5.0.x 
.editorconfig
```
dotnet_diagnostic.CA1051.severity = warning
dotnet_code_quality.CA1051.exclude_structs = true
dotnet_diagnostic.CA1826.severity = warning
dotnet_code_quality.CA1826.exclude_ordefault_methods = true
dotnet_diagnostic.CA2007.severity = warning
dotnet_code_quality.CA2007.exclude_async_void_methods = true
```
and only 2 warnings, not 0 or 3 show up
```
Warning: /home/runner/work/roslyn-analyzer-repro/roslyn-analyzer-repro/ra-repro1/repro1/Program.cs(10,29): warning CA1051: Do not declare visible instance fields [/home/runner/work/roslyn-analyzer-repro/roslyn-analyzer-repro/ra-repro1/repro1/repro1.csproj]
Warning: /home/runner/work/roslyn-analyzer-repro/roslyn-analyzer-repro/ra-repro1/repro1/Program.cs(17,21): warning CA1826: Do not use Enumerable methods on indexable collections. Instead use the collection directly. [/home/runner/work/roslyn-analyzer-repro/roslyn-analyzer-repro/ra-repro1/repro1/repro1.csproj]
```

How do I get the newer analyzer?

**Comment by wzchua** [comment]
I considered this not an issue anymore as i can move to .NET6
